rust/xaynet-analytics/src/data_provision/analytics_event.rs

```diff
@@ -20,13 +20,14 @@ impl AnalyticsEvent {
     pub fn new(
```

Comment:
Two little tricks that make using the API a bit easier `<N: Into<String>, R: Into<Option<String>>>`

```rust
impl AnalyticsEvent {
    pub fn new<N: Into<String>, R: Into<Option<String>>>(
        name: N,
        event_type: AnalyticsEventType,
        timestamp: DateTime<Utc>,
        screen_route: R,
    ) -> AnalyticsEvent {
        AnalyticsEvent {
            name: name.into(),
            event_type,
            timestamp,
            screen_route: screen_route.into(),
        }
    }
}
```

This way we can pass both `&str` as well as `String` in the case of `name` .
For `screen_route` we can omit the keyword `Some`.

```rust
        AnalyticsEvent::new(
            "test2".to_string(),
            AnalyticsEventType::ScreenEnter,
            timestamp,
            "other_screen".to_string(),
        );

        AnalyticsEvent::new(
            "test2",
            AnalyticsEventType::ScreenEnter,
            timestamp,
            None,
        );
```

Just a suggestion you don't have to implement this

Reply:
will do! looks good. it's the kind of stuff i still need to learn. thanks a lot